### PR TITLE
[DOC] procedures for manually triggering rolling updates

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

Procedures to describe manual rolling update of Kafka and ZooKeeper clusters using `StatefulSet` and `Pod` annotations. 

- Current content wrapped into new assembly and moved under _Managing Strimzi_ section in _Using Guide_
- Procedures split by annotation rather than cluster type: one for  `StatefulSet` and one for  `Pod` .
- Prerequisites move up a level, as they are same for both procedures. (We could remove if not necessary)

image

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/assemblies/managing/assembly-rolling-updates.adoc [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// This assembly is included in the following assemblies:
+//
+// using/assembly-management-tasks.adoc
+
+[id='assembly-rolling-updates-{context}']
+= Performing manual rolling updates of Kafka and ZooKeeper clusters
```
Comment (scholzj) [inline documentation/assemblies/managing/assembly-rolling-updates.adoc:6]: At least for me as a non-native speaker this is confusing. The _Perform manual rolling updates_ I would expect a guide how I can do them manually. But instead, the guide is about how to trigger them only. So maybe we can call it. _Trigger manual rolling updates ..._ or _Request ..._ or _Start ..._.

File: documentation/assemblies/managing/assembly-rolling-updates.adoc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Strimzi supports the use of annotations on `StatefulSet` and `Pods` resources to manually trigger a rolling update of Kafka and ZooKeeper clusters through the Cluster Operator.
+Rolling updates replace the pods of the resource with new ones, with no downtime.
```
Comment (scholzj) [inline documentation/assemblies/managing/assembly-rolling-updates.adoc:9]: The downtime depends on many factors - including topic configuration etc. So I would leave that part out. I also wonder if _restart the pods_ would fit better then _replace_.

File: documentation/assemblies/managing/assembly-rolling-updates.adoc [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// This assembly is included in the following assemblies:
+//
+// using/assembly-management-tasks.adoc
+
+[id='assembly-rolling-updates-{context}']
+= Performing manual rolling updates of Kafka and ZooKeeper clusters
+
+Strimzi supports the use of annotations on `StatefulSet` and `Pods` resources to manually trigger a rolling update of Kafka and ZooKeeper clusters through the Cluster Operator.
```
Comment (scholzj) [inline documentation/assemblies/managing/assembly-rolling-updates.adoc:8]: For some reason, _`Pods` resources_ sounds weird. Can it be just _`Pod` resources_ with Pod in singular?

**Review by scholzj — COMMENTED** [review]
Some nits. LGTM otherwise.

File: documentation/modules/managing/proc-manual-rolling-update-pods.adoc [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+= Performing a rolling update using a Pod annotation
+
+This procedure describes how to manually trigger a rolling update of an existing Kafka cluster or ZooKeeper cluster using a Kubernetes `Pod` annotation.
+When multiple pods from the same `StatefulSet` are annotated, consecutive rolling updates are performed within the same reconciliation run.
+
+.Procedure
+
+. Find the name of the Kafka or ZooKeeper `Pod` you want to manually update.
++
+For example, if your Kafka cluster is named _my-cluster_, the corresponding `Pod` names are _my-cluster-Kafka-index or_ and _my-cluster-zookeeper-index_.
```
Comment (ppatierno) [inline documentation/modules/managing/proc-manual-rolling-update-pods.adoc:15]: lower case "k" in the Kafka pod name

**Review by ppatierno — APPROVED** [review]
LGTM other then just a nit and comments from Jakub

File: documentation/modules/managing/proc-manual-rolling-update-statefulset.adoc [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+// managing/assembly-rolling-updates.adoc
+
+[id='proc-manual-rolling-update-statefulset-{context}']
+= Performing a rolling update using a StatefulSet annotation
+
+This procedure describes how to manually trigger a rolling update of an existing Kafka cluster or ZooKeeper cluster using a Kubernetes `StatefulSet` annotation.
+
+.Procedure
+
+. Find the name of the `StatefulSet` that controls the Kafka or ZooKeeper pods you want to manually update.
```
Comment (laidan6000) [inline documentation/modules/managing/proc-manual-rolling-update-statefulset.adoc:12]: The recommended style is StatefulSet without the monospace, unless you are referring to a specific resource. For example: https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/

File: documentation/modules/managing/proc-manual-rolling-update-pods.adoc (hunk 4 above)
Comment (laidan6000) [reply documentation/modules/managing/proc-manual-rolling-update-pods.adoc:15]: Typo: or and

File: documentation/modules/managing/proc-manual-rolling-update-pods.adoc [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
++
+[source,shell,subs=+quotes]
+----
+kubectl annotate pod _cluster-name_-kafka-_index_ strimzi.io/manual-rolling-update=true
+
+kubectl annotate pod _cluster-name_-zookeeper-_index_ strimzi.io/manual-rolling-update=true
+----
+
+. Wait for the next reconciliation to occur (every two minutes by default).
+A rolling update of all pods within the annotated `Pod` is triggered, as long as the annotation was detected by the reconciliation process.
```
Comment (laidan6000) [inline documentation/modules/managing/proc-manual-rolling-update-pods.adoc:30]: Pods within pods?

File: documentation/modules/managing/proc-manual-rolling-update-pods.adoc [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+When the rolling update of all the pods is complete, the annotation is removed from the `Pod`.
```
Comment (laidan6000) [inline documentation/modules/managing/proc-manual-rolling-update-pods.adoc:31]: Perhaps a command for annotating multiple pods would be helpful. Probably a question for @scholzj .

**Review by laidan6000 — CHANGES_REQUESTED** [review]
The manual rolling update procedure work well in the Managing section.

I added a few comments.

File: documentation/modules/managing/proc-manual-rolling-update-pods.adoc (hunk 7 above)
Comment (scholzj) [reply documentation/modules/managing/proc-manual-rolling-update-pods.adoc:31]: I do not think that exists. You just annotate them one by one.

File: documentation/assemblies/managing/assembly-rolling-updates.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/assemblies/managing/assembly-rolling-updates.adoc:6]: _Manually starting_?

File: documentation/assemblies/managing/assembly-rolling-updates.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/assemblies/managing/assembly-rolling-updates.adoc:9]: changed to _restart_ and removed downtime.

File: documentation/modules/managing/proc-manual-rolling-update-statefulset.adoc (hunk 5 above)
Comment (PaulRMellor) [reply documentation/modules/managing/proc-manual-rolling-update-statefulset.adoc:12]: We're referring to resources

File: documentation/modules/managing/proc-manual-rolling-update-pods.adoc (hunk 6 above)
Comment (PaulRMellor) [reply documentation/modules/managing/proc-manual-rolling-update-pods.adoc:30]: Changed to:

_A rolling update of the annotated `Pod` is triggered, as long as the annotation was detected by the reconciliation process. When the rolling update of a pod is complete, the annotation is removed from the `Pod`._

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
@laidan6000 Were your comments addressed? Can this be merged?

**Review by laidan6000 — APPROVED** [review]
Changes look good, thank you.

From a style perspective, we should discuss whether to adopt the Kubernetes standard of _not_ using monospace for resource names, apart from when referring to a specific resource in a procedure. For example, see [Service](https://kubernetes.io/docs/concepts/services-networking/service/).

**Comment by scholzj** [comment]
> From a style perspective, we should discuss whether to adopt the Kubernetes standard of not using monospace for resource names, apart from when referring to a specific resource in a procedure. For example, see Service.

I think we use that without consistency. Sometimes we use monospace, but also when using it without monospace we sometimes use them capitalized and sometimes not ... e.g. Services vs. services etc. (That said, the linked Kube doc is also inconsistent and uses monospaced `Service` on many places.) But I guess that can be left for another PR since this is in all docs and not just here.

**Comment by laidan6000** [comment]
Yep, please leave it until another discussion / PR. Using initial capitals can be part of the same discussion.

**Pull request merged**